src/ts/components/DashSocketIO.tsx

```diff
@@ -1,64 +1,95 @@
 import React from 'react';
 import { io } from 'socket.io-client';
-import {DashComponentProps} from '../props';
+import { DashComponentProps } from '../props';
+
+type SocketOptions = {
+  /** The socket.io endpoint path, defaults to '/socket.io' */
+  path?: string;
+  /** Whether to force a new connection */
+  forceNew?: boolean;
+  /** Timeout in milliseconds */
+  timeout?: number;
+  /** Whether to automatically connect on creation */
+  autoConnect?: boolean;
+  /** Additional query parameters */
+  query?: Record<string, any>;
+  /** Authentication data */
+  auth?: Record<string, any>;
+  [key: string]: any;
+};
 
 type Props = {
   /** The socket.io namespace url, defaults to window.location.origin */
-  url?: string
+  url?: string;
+  /** Socket.IO connection options */
+  options?: Partial<SocketOptions>;
   /** Name of Socket.IO events to listen to */
-  eventNames?: string[]
+  eventNames?: string[];
   /** Whether the client is connected to the websocket - READONLY */
-  connected?: boolean
+  connected?: boolean;
   /** The socket ID */
-  socketId?: string
+  socketId?: string;
 } & DashComponentProps;
+
+const defaultOptions: SocketOptions = {
+  path: '/socket.io',
+  forceNew: false,
+  autoConnect: true
+};
```

Comment:
Are these the default values if left blank? If so let's not pass anything.

Reply:
I am not sure if I completely understand. The defaultOptions set certain values for some of the variables.
Where some other props for the Socket, are not set by default. 

What part do you want to remove?

Reply:
the path forceNew and autoConnect already default to these values when nothing is passed to `io()` so no need to explicitly set them